src/net/http/request.go

```diff
@@ -1071,11 +1080,14 @@ func readRequest(b *bufio.Reader, deleteHostHeader bool) (req *Request, err erro
 	}
 
 	// Subsequent lines: Key: value.
-	mimeHeader, err := tp.ReadMIMEHeader()
+	mimeHeader, orderedHeader, err := tp.CFReadMIMEHeader()
 	if err != nil {
 		return nil, err
 	}
 	req.Header = Header(mimeHeader)
+	if cfRecordRequestLines {
```

Comment:
The global `cfRecordRequestLines` variable has to be removed and it needs to somehow use the `net/http.Server` field.

Also just set `CFHeaderLines` unconditionally here `CFReadMIMEHeader` *should* return `nil` when the feature is not enabled.

Reply:
`cfRecordRequestLines` isn't a global variable here, it's a parameter of function `readRequest`. The parameter *is* set to a global variable when called by `ReadRequest`, but it's set by `Server.CFRecordRequestLines` when called by the server.

Reply:
> The parameter is set to a global variable when called by ReadRequest

Yes that is exactly what I mean, you introduce a new global:
```diff
 // only supports HTTP/1.x requests. For HTTP/2, use golang.org/x/net/http2.
 func ReadRequest(b *bufio.Reader) (*Request, error) {
-       return readRequest(b, deleteHostHeader)
+       return readRequest(b, deleteHostHeader, cfRecordRequestLines)
 }

-// Constants for readRequest's deleteHostHeader parameter.
+// Constants for readRequest's flags.
 const (
        deleteHostHeader     = true
        keepHostHeader       = false
+       cfRecordRequestLines = false
 )
```

However after looking at it again, its sole purpose appears to be documentation. In `server.go`, `conn.readRequest` does propagate the `Server.CFRecordRequestLines` flag:

```diff
-       req, err := readRequest(c.bufr, keepHostHeader)
+       req, err := readRequest(c.bufr, keepHostHeader, c.server.CFRecordRequestLines)
        if err != nil {
```

Consider this concern resolved.